### ISSUE TITLE
Support md files

### ISSUE THREAD

**Issue by meri-burgess** [sphinx-contrib/emojicodes#40]
Currently, emojis are only supported in .rst. It would be great to be able to use the tool for .md files as well

**Comment by Peque** [comment]
Thanks for your suggestion @meri-burgess! :blush: 

Could you elaborate a bit more on your use case? I'm guessing you use an extension in Sphinx to support Markdown. If so, which one? Does it support variable substitutions like `|release|`, `|version|`?

**Comment by casperdcl** [comment]
In markdown, you can use [`eval-rst`](https://myst-parser.readthedocs.io/en/latest/syntax/roles-and-directives.html#syntax-directives-parsing):

````md
```{eval-rst}
|:white_check_mark:| hacky
```
````

to get:

✅ hacky

**Comment by jdillard** [comment]
I'm not positive, but it seems the extension only parses `.rst`:
https://github.com/sphinx-contrib/emojicodes/blob/ec8f49dcefae36ac36ba7100f8c6211b1c39819d/sphinxemoji/sphinxemoji.py#L55

where as it should support everything in [source_suffix](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-source_suffix)?

**Comment by jdillard** [comment]
myst-parser automatically adds `.md` to `source_suffx`: https://github.com/executablebooks/MyST-Parser/blob/ebf989fdae10c68dfbd3dcdfec4b86259b810cd9/myst_parser/sphinx_ext/main.py#L34

The problem isn't the syntax to be parsed, its the suffix (extension) of the files to be parsed.

**Comment by casperdcl** [comment]
yes I know, I was just commenting on what the future PR should probably support in terms of UX :)

**Comment by dannybeckett** [comment]
I am interested in trying to completely swap `rst` for `md` (I have no `.rst` files).

I tried this:

```
apt-get update && apt-get install git -y
git clone https://github.com/sphinx-contrib/emojicodes.git ../emojicodes
sed -i "s/rst/md/" ../emojicodes/sphinxemoji/sphinxemoji.py
cat ../emojicodes/sphinxemoji/sphinxemoji.py
pip install build
python -m build ../emojicodes
tree ../emojicodes
pip install ../emojicodes/dist/*.whl
```

It successfully replaces `rst` with `md` + loads the extension in to Sphinx, but unfortunately still doesn't want to display emojis, using any of `:fire:` or `|:fire:|` or `[:fire:]`

Any help is much appreciated please!

See also: [Stack Overflow question](https://stackoverflow.com/questions/42087466/sphinx-extension-to-use-github-markdown-emoji-in-sphinx)

Many thanks

**Comment by Peque** [comment]
I haven't had the time to look at this, but I'd be happy to review and integrate a PR if someone is able to implement it. :blush:
